### PR TITLE
Convert IP networks on Security Group resources to use IP Typing

### PR DESCRIPTION
**Pull request by ocrawford555**
This PR adds more accurate typing for the `CidrIp` and `CidrIpv6` fields that are contained within a Security Group model. 

Some tests were updated to reflect that Pydantic will now flag up invalid IP Network values passed to it.

### CODE REVIEW

File: pycfmodel/model/resources/properties/security_group_egress_prop.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 from .property import Property
 
 
+import ipaddress
+
+
 class SecurityGroupEgressProp(Property):
     CidrIp: Optional[ResolvableStr] = None
     CidrIpv6: Optional[ResolvableStr] = None
```
Comment (ocrawford555) [inline pycfmodel/model/resources/properties/security_group_egress_prop.py]: We might be able to change this to be of type IPv4 and IPv6 address, although unsure if this would have other side effects: https://pydantic-docs.helpmanual.io/usage/types/

Would it look like `CidrIp: Optional[ResolvableStrOrIpv4] = None` where `ResolvableStrOrIpv4` = `Resolvable[Union[str, ipaddress.Ipv4Address]]`

File: Makefile [hunk 2]
```diff
@@ -7,10 +7,10 @@ install-dev: install
 	pip install -e ".[dev]"
 
 format:
-	black .
+	black --exclude venv/ .
```
Comment (jsoucheiron) [inline Makefile]: you can add excluded paths in pyproject.toml

File: tests/resources/test_security_group_egress.py [hunk 3]
```diff
@@ -18,13 +18,13 @@
...
 
 @pytest.fixture()
-def security_group_egress_1():
+def security_group_egress_ipv4_1():
     return SecurityGroupEgress(
         **{
             "Type": "AWS::EC2::SecurityGroupEgress",
             "Properties": {
                 "GroupId": "sg-12341234",
-                "CidrIp": "1.1.1.1/0",
```
Comment (jsoucheiron) [inline tests/resources/test_security_group_egress.py]: why this change?

Comment (ocrawford555) [reply tests/resources/test_security_group_egress.py]: `1.1.1.1/0` is not a valid IP address in this new model:

```
E           pydantic.error_wrappers.ValidationError: 2 validation errors for SecurityGroupIngress
E           Properties -> CidrIp
E             value is not a valid IPv4 network (type=value_error.ipv4network)
E           Properties -> CidrIp
E             value is not a valid dict (type=type_error.dict)
```

File: pycfmodel/model/resources/properties/security_group_egress_prop.py [hunk 4]
```diff
@@ -29,11 +29,11 @@ class SecurityGroupEgressProp(Property):
     ToPort: Optional[ResolvableInt] = None
 
     def ipv4_slash_zero(self) -> bool:
-        if not self.CidrIp or not isinstance(self.CidrIp, str):
+        if not self.CidrIp:
             return False
-        return self.CidrIp.endswith("/0")
+        return str(self.CidrIp).endswith("/0")
```
Comment (oscarbc96) [inline pycfmodel/model/resources/properties/security_group_egress_prop.py]: What do you think about using host mask?
```suggestion
        return self.CidrIp.hostmask == IPv4Address('0.0.255.255')
```

Comment (ocrawford555) [reply pycfmodel/model/resources/properties/security_group_egress_prop.py]: Would it not be '255.255.255.255' mask value?

Comment (jsoucheiron) [reply pycfmodel/model/resources/properties/security_group_egress_prop.py]: oops: 
image

Comment (jsoucheiron) [reply pycfmodel/model/resources/properties/security_group_egress_prop.py]: This might break some things :(

Comment (jsoucheiron) [reply pycfmodel/model/resources/properties/security_group_egress_prop.py]: image

Comment (ocrawford555) [reply pycfmodel/model/resources/properties/security_group_egress_prop.py]: Added custom pre validator to use `strict=False` :)

File: pycfmodel/model/resources/properties/security_group_egress_prop.py [hunk 5]
```diff
@@ -12,28 +12,39 @@
...
+    @validator("CidrIpv6", pre=True)
+    def set_CidrIpv6(cls, v):
+        return IPv6Network(v, strict=False)
+
     def ipv4_slash_zero(self) -> bool:
-        if not self.CidrIp or not isinstance(self.CidrIp, str):
+        if not self.CidrIp:
             return False
-        return self.CidrIp.endswith("/0")
+        return self.CidrIp.hostmask == IPv4Address(IPV4_MASK_VALUE)
```
Comment (jsoucheiron) [inline pycfmodel/model/resources/properties/security_group_egress_prop.py]:
```suggestion
        return self.CidrIp == IPv4Address('0.0.0.0/0')
```

File: pycfmodel/model/resources/properties/security_group_egress_prop.py [hunk 6]
```diff
@@ -12,28 +12,39 @@
...
 
     def ipv6_slash_zero(self) -> bool:
-        if not self.CidrIpv6 or not isinstance(self.CidrIpv6, str):
+        if not self.CidrIpv6:
             return False
-        return self.CidrIpv6.endswith("/0")
+        return self.CidrIpv6.hostmask == IPv6Address(IPV6_MASK_VALUE)
```
Comment (jsoucheiron) [inline pycfmodel/model/resources/properties/security_group_egress_prop.py]:
```suggestion
        return self.CidrIpv6 == IPv6Address('::/0')
```

File: pycfmodel/model/resources/properties/security_group_ingress_prop.py [hunk 7]
```diff
@@ -30,12 +33,20 @@ class SecurityGroupIngressProp(Property):
     SourceSecurityGroupOwnerId: Optional[ResolvableStr] = None
     ToPort: Optional[ResolvableInt] = None
 
+    @validator("CidrIp", pre=True)
+    def set_CidrIp(cls, v):
+        return IPv4Network(v, strict=False)
```
Comment (jsoucheiron) [inline pycfmodel/model/resources/properties/security_group_ingress_prop.py]: Make sure it deals with resolvables too

Comment (ocrawford555) [reply pycfmodel/model/resources/properties/security_group_ingress_prop.py]: Updated resolver test to show it using `strict=False`
